pkg/frontend/admin_vm_actions.go

```diff
@@ -0,0 +1,80 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+type VMContext struct {
+	context context.Context
+	r       *http.Request
+	logger  *logrus.Entry
+}
+
+type AzActionsFactoryWrap struct {
```

Comment:
You don't need to export this as it's only used within the package. 

Additionally this contains VM specific parameters and not all our azure actions contain are against VMs.  We have other things that run against NICs instead of strictly Vms.  I think renaming it to something like `azVmActionsWrapper` or `azVmActionsFactoryWrap` would be better.  

Or, wrap all fields that would be needed and replace it in multiple places.   (not just the azure VM actions)